migration/idempotent/001-base.sql

```diff
@@ -2563,7 +2563,7 @@ LANGUAGE PLPGSQL;
 REVOKE ALL ON FUNCTION _prom_catalog.create_metric_view(text) FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION _prom_catalog.create_metric_view(text) TO prom_writer;
 
-CREATE OR REPLACE FUNCTION prom_api.register_metric_view(schema_name text, view_name text, if_not_exists BOOLEAN = false)
+CREATE OR REPLACE FUNCTION prom_api.register_metric_view(schema_name text, view_name text, refresh_interval INTERVAL, rollup_id BIGINT = NULL, if_not_exists BOOLEAN = false)
```

Comment:
What are the implications for the `refresh_interval=NULL` is it only that customer will have to set their own refresh policies for the Caggs? Is there something preventing them from setting their custom config with the `startup.dataset` config?

Reply:
> refresh_interval=NULL is it only that customer will have to set their own refresh policies for the Caggs?

Yes, but we never recommend to supply `NULL` as an input. This is to avoid creating individual maintenance jobs per Cagg which hurts the performance. In the docs, we will mention `refresh_interval` as a required parameter. Supplying `NULL` will be an exceptional case where someone does not want maintenance to be handled automatically.

> setting their custom config with the startup.dataset config?

I don't think users can control maintenance things from `startup.dataset` config.